test/connection.ts

```diff
@@ -619,4 +619,13 @@ describe('Connection', function () {
       .then(() => new Error('Should not have succeeded'))
       .catch(done())
   })
+
+  it('should throw error if pending response with same ID', async function () {
+    await this.client.connection.request({ id: 'test', command: 'ping' })
```

Comment:
`await`-ing this should remove that request from the `pending` responses, right?

You should be able to make as many promises w/ the same `id` as you want, as long as you await them one at a time. 

IMO this should only throw if both promises are `pending` AND have the same `id`.